File: quiz/src/main/java/com/example/quiz/service/quiz_topic/QuizTopicService.java
package com.example.quiz.service.quiz_topic;

import com.example.quiz.base.baseInterface.BaseService;
import com.example.quiz.model.entity.quiz_topic.QuizTopicRequestDto;
import com.example.quiz.model.entity.quiz_topic.QuizTopicResponseDto;
import com.example.quiz.model.entity.quiz_topic.QuizTopic;
import com.example.quiz.model.entity.quiz_topic.QuizTopicView;

import java.util.List;

public interface QuizTopicService extends BaseService<QuizTopic, Long, QuizTopicRequestDto, QuizTopicResponseDto, QuizTopicView> {
}

File: quiz-fe/src/share/services/quiz_topic/quiz-topic.service.ts
import { ENV } from '../../utils/env';
import { BaseService } from "../BaseService";
import { QuizTopic, QuizTopicRequest, QuizTopicResponse, QuizTopicView } from './model';

export class QuizTopicService extends BaseService<
QuizTopic,
QuizTopicRequest,
QuizTopicResponse,
QuizTopicView
> {
     constructor() {
          super(ENV.API_URL, 'quiz-topics');
     }

     async submit(id: string | number, answers: Record<number, number>): Promise<any> {
          return this.handleRequest<any>(`${this.getEndpoint()}/${id}/submit`, {
               method: 'POST',
               body: JSON.stringify({ answers }),
          });
     }
}

export const quizTopicService = new QuizTopicService();